src/Confluent.SchemaRegistry.Serdes.Avro/AvroDeserializer.cs

```diff
@@ -35,7 +35,7 @@ namespace Confluent.SchemaRegistry.Serdes
     ///       bytes 1-4:        Unique global id of the Avro schema that was used for encoding (as registered in Confluent Schema Registry), big endian.
     ///       following bytes:  The serialized data.
     /// </remarks>
-    public class AvroDeserializer<T> : IAsyncDeserializer<T>
+    public class AvroDeserializer<T> : IAsyncDeserializer<T> where T : class
```

Comment:
Assuming `class` implies that the type is nullable?

Reply:
yes. but this won't compile because we explicitly handle the `int` etc. cases. even if we didn't, it's not a backwards compatible change. it's likely people rely on this, since we talk about it in places, and I don't think we can change to `int?` until a major version bump. straightforward for people to update to, but it'll break things.

Reply:
How about providing a `ProduceTombstone(key)` and a `Message.IsTombstone()` ?

Reply:
i don't think we want to go adding to the core client APIs just to get around this - would prefer to add a second set of avro serdes.

Reply:
Tombstones/compaction is a core Kafka feature, so I believe this abstraction should indeed be in the core clients rather than the serdes.
Tombstones being represented as Null is a protocol implementation detail and not necessarily something that needs to be exposed the same way to the application.

Reply:
true, but we'd be exposing the same information redundantly (which I like less).

also, this does not actually do anything to address the problem which is at the serde level, not the client level.